indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -804,6 +805,23 @@ public SeekableStreamSupervisor(
               : Math.min(10, this.ioConfig.getTaskCount() * this.ioConfig.getReplicas()));
     }
 
+    IdleConfig specIdleConfig = spec.getIoConfig().getIdleConfig();
+    if (specIdleConfig != null) {
+      if (specIdleConfig.getInactiveAfterMillis() != null) {
```

Comment:
I feel we might still incorrectly handle the case where user has specified `inactiveAfterMillis` at the supervisor level but not `enabled`. In that case, we should just override the `inactiveAfterMillis` and take the cluster level value for `enabled`.

But I guess that's a corner case and we can do it in a later PR.

Reply:
That is not done deliberately since KinesisSupervisor should not go idle when cluster-level config is enabled. It can be done with KinesisSupervisor PR.